### DISCUSSION TITLE
Home Assistant 400 Error

### DISCUSSION THREAD

**Discussion by zypherion** [warp-tech/warpgate#747] [Q&A] [answered]
Hi All,
Just found warpgate and it looks to be the solution for a major pain point for me with my home server. I've got a few services setup and working already but I'm hitting a snag trying to reach Home Assistant through warp gate.

When I try to reach home assistant I'm getting a 400 Bad request error in my browser and in the home assistant logs I get:
**ERROR (MainThread) [homeassistant.components.http.forwarded] Invalid IP address in X-Forwarded-For: 192.168.1.10:51036**

That is the correct IP address of my warpgate and I've already added that IP to my trusted proxies. However I read elsewhere that home assistant needs to get the X-Forwarded-For address without a trailing port number. Importantly, the port is changing on each attempt. Is there any way that's possible with warpgate or no?

**Comment by Eugeny** [comment, marked as answer]
Fixed! The new nightly build should be ready in a few min: https://github.com/warp-tech/warpgate/actions/runs/4788374375

**Comment by ecki** [comment]
I was able to get the 400 fixed, by configuring the trusted proxy ip. However the web app is now constantly reloading (in safari). In the logs I see a successful websocket established, but maybe that one is constantly aborted which causes the reload?
